src/openloco/companymgr.cpp

```diff
@@ -113,4 +119,76 @@ namespace openloco::companymgr
             }
         }
     }
+
+    // 0x004383ED
+    void updateOwnerStatus()
+    {
+        if (openloco::is_title_mode() || openloco::is_editor_mode())
+        {
+            return;
+        }
+
+        auto company = companymgr::get(_updating_company_id);
+        if (company == nullptr)
+        {
+            return;
+        }
+
+        company->var_12 += 1;
+        if ((company->var_12 & 0x7F) != 0)
+            return;
+
+        for (size_t i = 0; i < WindowManager::count(); i++)
+        {
+            auto w = WindowManager::get(i);
+
+            if (w->type != WindowType::vehicle)
+                continue;
+
+            auto vehicle = thingmgr::get<openloco::vehicle>(w->number);
+            if (vehicle->x == -0x8000)
+                continue;
+
+            if (vehicle->owner != _updating_company_id)
+                continue;
+
+            game_commands::do_73(vehicle->id);
+            return;
+        }
+
+        auto main = WindowManager::getMainWindow();
+        if (main == nullptr)
+            return;
+
+        auto viewport = main->viewports[0];
+        if (viewport == nullptr)
+            return;
+
+        gfx::point_t screenPosition;
+        screenPosition.x = viewport->x + viewport->width / 2;
+        screenPosition.y = viewport->y + viewport->height / 2;
+
+        registers r1;
+        r1.ax = screenPosition.x;
+        r1.bx = screenPosition.y;
+        call(0x0045F1A7, r1);
+        ui::viewport* vp = (ui::viewport*)r1.edi;
+        auto mapPosition = map::map_pos(r1.ax, r1.bx);
+
+        // Happens if center of viewport is obstructed. Probably estimates the centre location
+        if (mapPosition.x == -0x8000 || viewport != vp)
+        {
+            registers r2;
+
+            r2.ax = viewport->view_x + viewport->view_width / 2;
+            r2.bx = viewport->view_y + viewport->view_height / 2;
+            r2.edx = viewport->getRotation();
+            call(0x0045F997, r2);
+
+            mapPosition.x = r2.ax;
+            mapPosition.y = r2.bx;
+        }
```

Comment:
Its interesting that there is a call to `map_element_height` in the else branch but then it just clears the result of the call.